### PR TITLE
Add com.lablicate.OpenChrom

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/OpenChrom/openchrom/issues/435
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: com.lablicate.OpenChrom.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    build-commands:
+      - mv openchrom ${FLATPAK_DEST}/openchrom/
+      - mkdir -p ${FLATPAK_DEST}/bin
+      - ln -sf ${FLATPAK_DEST}/openchrom/openchrom ${FLATPAK_DEST}/bin/
+      - install -Dm644 ${FLATPAK_ID}.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
+      - install -Dm644 ${FLATPAK_ID}.png ${FLATPAK_DEST}/share/icons/hicolor/256x256/apps/${FLATPAK_ID}.png
+      - install -Dm644 ${FLATPAK_ID}.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml
+    sources:
+      - type: archive
+        url: https://products.lablicate.com/openchrom/1.5.0/openchrom-lablicate_linux.x86_64_1.5.0.tar.gz
```
Comment (hfiguiere) [inline com.lablicate.OpenChrom.yaml:30]: so what happen on aarch64?

File: com.lablicate.OpenChrom.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  <description>
+    <p>OpenChrom® is a vendor independent open source software for chromatography, spectrometry and spectroscopy. It has a strong focus on chromatography and nominal mass spectrometry, but next to GC/MS and GC/FID it has basic support for HPLC-UV/VIS, FTIR, PCR and NMR.</p>
+    <p>The base version is free and open source software with most of it curated under the Eclipse ChemClipse umbrella. Its numerous carefully reverse-engineered file format converters for vendors like Agilent, Thermo, PerkinElmer, Shimadzu, Varian, Bruker, ChromTech, Thermo/Finnigan, Waters, CAMAG, ABSciex, MassFinder, EZChrom... are however closed source and have to be installed at runtime accepting the proprietary license terms.</p>
+    <p>Moreover the Lablicate marketplace allows for the installation of proprietary databases and plug-ins to speed up workflows. These are reserved for commercial customers. Students with a valid university e-Mail can apply for significant discounts on the Lablicate portfolio.</p>
+  </description>
+  <screenshots>
+    <screenshot type="default">
+      <image type="source">https://lablicate.com/static/media/openchrom.88bf4b9f.png</image>
```
Comment (hfiguiere) [inline com.lablicate.OpenChrom.metainfo.xml:17]: it's one `image` tag per `screenshot`. That's part of the build failure

it is also missing the `launchable`

File: com.lablicate.OpenChrom.yaml (hunk 1 above)
Comment (Mailaender) [reply com.lablicate.OpenChrom.yaml:33]: Good point: coming soon.

**Comment by Mailaender** [comment]
bot, build com.lablicate.OpenChrom

**Comment by Mailaender** [comment]
bot, build com.lablicate.OpenChrom

File: com.lablicate.OpenChrom.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,55 @@
+<?xml version='1.0' encoding='utf-8'?>
+<component type="desktop">
+  <id>com.lablicate.OpenChrom</id>
+  <name>OpenChrom</name>
+  <summary>Chromatography Data System</summary>
+  <developer_name>Lablicate GmbH</developer_name>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>EPL-1.0 AND LicenseRef-proprietary=https://lablicate.com/about/terms</project_license>
```
Comment (bbhtt) [inline com.lablicate.OpenChrom.metainfo.xml:8]: Is redistribution allowed under the license? Did you contact the upstream?

Comment (Mailaender) [reply com.lablicate.OpenChrom.metainfo.xml:8]: Yes, redistribution is allowed. Everything that is proprietary can be downloaded at runtime or is part of the internal distribution mechanism. The AppStream metadata file explains the concept. Note: this does not apply to older versions.

Comment (bbhtt) [reply com.lablicate.OpenChrom.metainfo.xml:8]: Where is this written and which part is under the proprietary license?

Appstream doesn't matter.

Unless explicitly written that redistribution is allowed, proprietary licensed stuff needs to use extra data

File: com.lablicate.OpenChrom.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+command: openchrom
+finish-args:
+  - --socket=x11
+  - --share=ipc
+  - --device=dri
+  - --share=network
+  - --socket=pulseaudio
+  - --persist=.openchrom
+  - --persist=.eclipseswtchartsettings
+  - --filesystem=/tmp:rw
```
Comment (bbhtt) [inline com.lablicate.OpenChrom.yaml:14]: Why is this needed? Can't you set the TMPDIR env var?

Comment (Mailaender) [reply com.lablicate.OpenChrom.yaml:14]: Taken from https://github.com/flathub/org.eclipse.Java/blob/49768643348b2efcfa330eb0fe41df5317ba9d21/org.eclipse.Java.json#L11 it is required for the p2 update mechanism.

File: com.lablicate.OpenChrom.metainfo.xml (hunk 3 above)
Comment (Mailaender) [reply com.lablicate.OpenChrom.metainfo.xml]: It was simpler to remove all proprietary bits and download them at runtime inside the application instead. https://github.com/OpenChrom/openchrom/pull/481

**Comment by Mailaender** [comment]
bot, build com.lablicate.OpenChrom

**Comment by bbhtt** [comment]
bot, build com.lablicate.OpenChrom

**Comment by bbhtt** [comment]
Can icon, desktop file and metainfo be submitted upstream?

**Comment by hfiguiere** [comment]
Checksums are incorrect for the binary tarballs.

**Comment by Mailaender** [comment]
> Can icon, desktop file and metainfo be submitted upstream?

It is a Maven/Tycho build that creates the tarballs, so that would be difficult.

**Comment by Mailaender** [comment]
bot, build com.lablicate.OpenChrom

**Comment by bbhtt** [comment]
```
flatpak run com.lablicate.OpenChrom
qemu-aarch64-static: Could not open '/lib/ld-linux-aarch64.so.1': No such file or directory
```

**Comment by Mailaender** [comment]
Are you sure that it is related to this package?

**Comment by bbhtt** [comment]
Seems to be, @hfiguiere do you see the same on your end

**Comment by hfiguiere** [comment]
looks like you are trying to run the aarch64 with qemu and it can't find ld.so.... not sure how you reached that point

**Comment by bbhtt** [comment]
> looks like you are trying to run the aarch64 with qemu and it can't find ld.so.... not sure how you reached that point

Is the test build working on your end?

**Comment by bbhtt** [comment]
bot, build com.lablicate.OpenChrom

**Comment by bbhtt** [comment]
bot, build com.lablicate.OpenChrom

File: com.lablicate.OpenChrom.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
+app-id: com.lablicate.OpenChrom
+runtime: org.freedesktop.Platform
+runtime-version: 23.08
```
Comment (bbhtt) [inline com.lablicate.OpenChrom.yaml:3]:
```suggestion
runtime-version: "23.08"

```

Comment (bbhtt) [reply com.lablicate.OpenChrom.yaml:3]: image

File: com.lablicate.OpenChrom.yaml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      - mv openchrom ${FLATPAK_DEST}/openchrom/
+      - mkdir -p ${FLATPAK_DEST}/bin
+      - ln -sf ${FLATPAK_DEST}/openchrom/openchrom ${FLATPAK_DEST}/bin/
+      - install -Dm644 ${FLATPAK_ID}.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
+      - install -Dm644 ${FLATPAK_ID}.png ${FLATPAK_DEST}/share/icons/hicolor/256x256/apps/${FLATPAK_ID}.png
+      - install -Dm644 ${FLATPAK_ID}.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml
+    sources:
+      - type: archive
+        only-arches:
+          x86_64
```
Comment (bbhtt) [inline com.lablicate.OpenChrom.yaml:32]:
```suggestion
      - type: archive
        only-arches:
          - x86_64

```

File: com.lablicate.OpenChrom.yaml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        url: https://products.lablicate.com/openchrom/1.5.0/openchrom-lablicate_linux.x86_64_1.5.0.tar.gz
+        sha256: 64a8528d889f6a05dffc499206af04e93ee170814700c416183fdf42f4d000eb
+        strip-components: 0
+        dest: openchrom
+      - type: archive
+        only-arches:
+          aarch64
```
Comment (bbhtt) [inline com.lablicate.OpenChrom.yaml:39]:
```suggestion
      - type: archive
        only-arches:
          - aarch64

```

**Comment by bbhtt** [comment]
> Are you sure that it is related to this package?

@Mailaender The issue was `only-arches` being wrongly formatted, so builders where building the wrong package for wrong arch. Likely

File: com.lablicate.OpenChrom.yaml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+      - ln -sf ${FLATPAK_DEST}/openchrom/openchrom ${FLATPAK_DEST}/bin/
+      - install -Dm644 ${FLATPAK_ID}.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
+      - install -Dm644 ${FLATPAK_ID}.png ${FLATPAK_DEST}/share/icons/hicolor/256x256/apps/${FLATPAK_ID}.png
+      - install -Dm644 ${FLATPAK_ID}.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml
+    sources:
+      - type: archive
+        only-arches:
+          x86_64
+        url: https://products.lablicate.com/openchrom/1.5.0/openchrom-lablicate_linux.x86_64_1.5.0.tar.gz
+        sha256: 64a8528d889f6a05dffc499206af04e93ee170814700c416183fdf42f4d000eb
```
Comment (bbhtt) [inline com.lablicate.OpenChrom.yaml:34]: Checksum changed.

**Comment by Mailaender** [comment]
bot, build com.lablicate.OpenChrom

**Comment by bbhtt** [comment]
bot, build com.lablicate.OpenChrom

File: com.lablicate.OpenChrom.yaml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+          - x86_64
+        url: https://products.lablicate.com/openchrom/1.5.0/openchrom-lablicate_linux.x86_64_1.5.0.tar.gz
+        sha256: 2fa1994e6434d9fd6878b009cb147013c45adce193952f906dfd60103fe21242
+        strip-components: 0
+        dest: openchrom
+      - type: archive
+        only-arches:
+          - aarch64
+        url: https://products.lablicate.com/openchrom/1.5.0/openchrom-lablicate_linux.aarch64_1.5.0.tar.gz
+        sha256: 7683ea34a14e9cc6ad3bab97b892832f924137d00dc3d005b3fa0e3c429a1dfe
```
Comment (bbhtt) [inline com.lablicate.OpenChrom.yaml:41]:
```suggestion
        sha256: 9074c510f0f2b265343c5ec5f2251b4cb020da0f21e2bb9b874517f7ecd983ce
```

File: com.lablicate.OpenChrom.yaml [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+      - ln -sf ${FLATPAK_DEST}/openchrom/openchrom ${FLATPAK_DEST}/bin/
+      - install -Dm644 ${FLATPAK_ID}.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
+      - install -Dm644 ${FLATPAK_ID}.png ${FLATPAK_DEST}/share/icons/hicolor/256x256/apps/${FLATPAK_ID}.png
+      - install -Dm644 ${FLATPAK_ID}.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml
+    sources:
+      - type: archive
+        only-arches:
+          - x86_64
+        url: https://products.lablicate.com/openchrom/1.5.0/openchrom-lablicate_linux.x86_64_1.5.0.tar.gz
+        sha256: 2fa1994e6434d9fd6878b009cb147013c45adce193952f906dfd60103fe21242
```
[the remark this answers is not in the document]
Comment (bbhtt) [reply com.lablicate.OpenChrom.yaml]: I'm updating the checksum, because it keeps going out of my radar.

**Comment by Mailaender** [comment]
bot, build com.lablicate.OpenChrom

**Comment by Mailaender** [comment]
Please add @eselmeister as he is the founder of the project.

**Comment by bbhtt** [comment]
> Please add @eselmeister as he is the founder of the project.

Added
